pkg/proxy/proxy.go

```diff
@@ -279,76 +233,89 @@ func (p *Proxy) handleFleetTelemetryConfig(host string, w http.ResponseWriter, r
-	defer cancel()
+func (p *Proxy) handleVehicleCommand(w http.ResponseWriter, req *http.Request) {
+	ctx := req.Context()
+	command := req.PathValue("command")
+	vin := req.PathValue("vin")
+	acct := req.Context().Value(contextKey("account")).(account.AccountService)
+
+	if len(vin) != vinLength {
+		writeJSONError(w, http.StatusNotFound, errors.New("expected 17-character VIN in path (do not use vehicle ID)"))
+		return
+	}
+	if p.isNotSupported(vin) {
+		p.forwardRequest(w, req)
+		return
+	}
 
 	// Serialize commands sent to a specific VIN to avoid some complexities associated with sharing
 	// the vehicle.Vehicle object. VCSEC commands fail if they arrive out of order, anyway.
 	if err := p.lockVIN(ctx, vin); err != nil {
 		writeJSONError(w, http.StatusServiceUnavailable, err)
-		return err
+		return
 	}
 	defer p.unlockVIN(vin)
 
-	car, commandToExecuteFunc, err := p.loadVehicleAndCommandFromRequest(ctx, acct, w, req, command, vin)
+	vehicle, commandToExecuteFunc, err := p.loadVehicleAndCommandFromRequest(ctx, acct, w, req, command, vin)
 	if err != nil {
-		return err
+		return
 	}
 
-	if err := car.Connect(ctx); err != nil {
+	if err := vehicle.Connect(ctx); err != nil {
 		writeJSONError(w, http.StatusInternalServerError, err)
-		return err
+		return
 	}
-	defer car.Disconnect()
+	defer vehicle.Disconnect()
 
-	if err := car.StartSession(ctx, nil); errors.Is(err, protocol.ErrProtocolNotSupported) {
+	if err := vehicle.StartSession(ctx, nil); errors.Is(err, protocol.ErrProtocolNotSupported) {
 		p.markUnsupportedVIN(vin)
-		p.forwardRequest(acct.Host, w, req)
-		return err
+		p.forwardRequest(w, req)
+		return
 	} else if err != nil {
 		writeJSONError(w, http.StatusInternalServerError, err)
-		return err
+		return
 	}
-	defer car.UpdateCachedSessions(p.sessions)
+	defer vehicle.UpdateCachedSessions(p.sessions)
 
-	if err = commandToExecuteFunc(car); err == ErrCommandUseRESTAPI {
-		return err
+	if err = commandToExecuteFunc(vehicle); err == ErrCommandUseRESTAPI {
+		p.forwardRequest(w, req)
+		return
 	}
 	if protocol.IsNominalError(err) {
 		writeJSONError(w, http.StatusOK, err)
-		return err
+		return
 	}
 	if err != nil {
 		writeJSONError(w, http.StatusInternalServerError, err)
-		return err
+		return
 	}
 
-	w.Header().Add("Content-Type", "application/json")
-	fmt.Fprintln(w, "{\"response\":{\"result\":true,\"reason\":\"\"}}")
-	return nil
+	writeJSON(w, http.StatusOK, []byte(`{"response":{"result":true,"reason":""}}`))
 }
 
-func (p *Proxy) loadVehicleAndCommandFromRequest(ctx context.Context, acct *account.Account, w http.ResponseWriter, req *http.Request,
-	command, vin string) (*vehicle.Vehicle, func(*vehicle.Vehicle) error, error) {
+func (p *Proxy) loadVehicleAndCommandFromRequest(ctx context.Context, acct account.AccountService, w http.ResponseWriter, req *http.Request,
+	command, vin string) (vehicle.VehicleService, func(vehicle.VehicleService) error, error) {
```

Comment:
hmm maybe beyond the scope of this PR but `account.AccountService` and `vehicle.VehicleService` doesn't seems to conform to golang style https://sben.dev/stutter/

Reply:
No longer introducing these interfaces but will be sure to keep this in mind in the future.